src/libopensc/card-iasecc.c

```diff
@@ -1232,7 +1232,7 @@ iasecc_process_fci(struct sc_card *card, struct sc_file *file,
 	else
 		acls = sc_asn1_find_tag(ctx, buf, buflen, IASECC_DOCP_TAG_ACLS_CONTACT, &taglen);
 
-	if (!acls || taglen < 7)   {
+	if (!acls)   {
```

Comment:
I think that this will still crash for acls that have  exactly one byte. This condition will be skipped and on line 1251 you will be accessing acls[1] unconditionally.